modules/casper.js

```diff
@@ -2204,6 +2213,105 @@ function getTimeoutAndCheckNextStepFunction(timeout, then, methodName, defaultTi
 }
 
 /**
+ * Makes the provided frame page as the currently active one. Note that the
+ * active page will NOT be reverted when finished.
+ *
+ * @param  String|Number frameInfo  Target frame name or number
+ * @return Casper
+ */
+Casper.prototype.switchToChildFrame = function switchToChildFrame(frameInfo) {
+    "use strict";
+    if (utils.isNumber(frameInfo)) {
+        if (frameInfo > this.page.childFramesCount() - 1) {
+            throw new CasperError(f('Frame number "%d" is out of bounds.', frameInfo));
+        }
+        this.page.context = this.getElementBounds(selectXPath('(//frame|//iframe)[' + (frameInfo + 1) + ']'), true);
+    } else if (this.page.childFramesName().indexOf(frameInfo) === -1) {
+        throw new CasperError(f('No frame named "%s" was found.', frameInfo));
+    } else {
+        this.page.context = this.getElementBounds(selectXPath('(//frame|//iframe)[@name="' + frameInfo + '"]'), true);
+    }
+    // make the frame page the currently active one
+    this.page.switchToFrame(frameInfo);
+
+
+    return this;
+};
+
+/**
+ * Makes the provided frame page as the currently active one. Note that the
+ * active page will NOT be reverted when finished.
+ *
+ * @param  String|Number frameInfo  Target frame name or number
+ * @return Casper
+ */
+Casper.prototype.switchToFrame = function switchToFrame(frameInfo) {
+    "use strict";
+    this.switchToChildFrame(frameInfo);
+    this.page.frameLevel++;
```

Comment:
Should add a `this.page.frameLevel--;` to `switchToParentFrame()` ?
And `this.page.frameLevel = 0;` to `switchToMainFrame()` ?